controllers/awsmachinetemplate_controller.go

```diff
@@ -492,3 +536,28 @@ func (r *AWSMachineTemplateReconciler) machineDeploymentToAWSMachineTemplate(ctx
 		},
 	}
 }
+
+// machineSetToAWSMachineTemplate maps MachineSet to AWSMachineTemplate reconcile requests.
+// This enables the controller to reconcile AWSMachineTemplate when its owner MachineSet is created or updated,
+// ensuring that nodeInfo can be populated even if the cache hasn't synced yet.
+func (r *AWSMachineTemplateReconciler) machineSetToAWSMachineTemplate(ctx context.Context, o client.Object) []ctrl.Request {
+	md, ok := o.(*clusterv1.MachineSet)
+	if !ok {
+		return nil
+	}
+
+	// Check if it references an AWSMachineTemplate
+	if md.Spec.Template.Spec.InfrastructureRef.Kind != awsMachineTemplateKind {
+		return nil
+	}
+
+	// Return reconcile request for the referenced AWSMachineTemplate
+	return []ctrl.Request{
+		{
+			NamespacedName: client.ObjectKey{
+				Namespace: md.Namespace,
+				Name:      md.Spec.Template.Spec.InfrastructureRef.Name,
+			},
+		},
+	}
+}
```

Comment:
```
func (r *AWSMachineTemplateReconciler) kubeadmControlPlaneToAWSMachineTemplate(ctx context.Context, o client.Object) []ctrl.Request
func (r *AWSMachineTemplateReconciler) machineDeploymentToAWSMachineTemplate(ctx context.Context, o client.Object) []ctrl.Request
func (r *AWSMachineTemplateReconciler) machineSetToAWSMachineTemplate(ctx context.Context, o client.Object) []ctrl.Request
```
These three functions are very similar. Should we refactor them a bit, similar to how we handle objectToAWSMachineTemplate?

Reply:
Imho not worth the complexity compared to these simple functions.